### ISSUE TITLE
Fail to Mix.install in case that Hex is not installed even if specifying the option force: true.

### ISSUE THREAD

**Issue by zacky1972** [elixir-lang/elixir#12086]
### Elixir and Erlang/OTP versions

Erlang/OTP 25 [erts-13.0.3] [source] [64-bit] [smp:20:20] [ds:20:20:10] [async-threads:1] [jit]

Elixir 1.14.0-rc.1 (185eeec) (compiled with Erlang/OTP 25)

### Operating system

macOS Monterey 12.5

### Current behavior

```elixir
> Mix.install([{:flow, "~>1.0"}], force: true)
Could not find Hex, which is needed to build dependency :flow
Shall I install Hex? (if running non-interactively, use "mix local.hex --force") [Yn] 
```

### Expected behavior

`Mix.install` will succeed without prompting.

**Comment by josevalim** [comment]
Closing in favor of #12086.

**Comment by zacky1972** [comment]
Thanks!

**Comment by lukaszsamson** [comment]
It seems the fix for that issue never made to 1.14 release. This problem is manifesting for one of ElixirLS users https://github.com/elixir-lsp/vscode-elixir-ls/issues/392
